src/hmc.py

```diff
@@ -0,0 +1,298 @@
+import logging
+
+from memory import Memory
+
+LOGGER = logging.getLogger(__name__)
```

Comment:
No need to instantiate this logging entity, just use the parent.logger

Reply:
Agreed. Fixed.